addons/hr/models/hr_employee.py

```diff
@@ -514,6 +514,18 @@ def _get_tz_batch(self):
         # Returns a dict {employee_id: tz}
         return {emp.id: emp._get_tz() for emp in self}
 
+    def _get_expected_attendances(self, date_from, date_to):
+        self.ensure_one()
```

Comment:
On a second chain of thought, shouldn't we make this method work in batch? Returning an object like `{employee: result for employee in self}`
We can sort employees by calendars and only call `_work_intervals_batch` once per calendar instead.
Not sure if we could improve performances that much on the calendar side though, but that's already that :D

Reply:
TODO master, batchify _work_intervals_batch better